wp-admin/js/widgets/text-widgets.js

```diff
@@ -0,0 +1,197 @@
+			if ( tinymce.get( id ) ) {
+				delete tinymce.editors[ id ];
+				tinymce.remove( '#' + id );
+			}
+
+			wp.editor.initialize( id, {
+				tinymce: {
+					wpautop: true
+				},
+				quicktags: true
+			} );
+
+			editor = window.tinymce.get( id );
+			editor.on( 'change', _.debounce( function() {
+				editor.save();
+				textarea.trigger( 'change' );
+			}, changeDebounceDelay ) );
+		}
+	});
+
+	/**
+	 * Mapping of widget ID to instances of TextWidgetControl subclasses.
+	 *
+	 * @type {Object.<string, wp.textWidgets.TextWidgetControl>}
+	 */
+	component.widgetControls = {};
+
+	/**
+	 * Handle widget being added or initialized for the first time at the widget-added event.
+	 *
+	 * @param {jQuery.Event} event - Event.
+	 * @param {jQuery}       widgetContainer - Widget container element.
+	 * @returns {void}
+	 */
+	component.handleWidgetAdded = function handleWidgetAdded( event, widgetContainer ) {
+		var widgetContent, widgetForm, idBase, widgetControl, widgetId, animatedCheckDelay = 50, widgetInside, renderWhenAnimationDone;
+		widgetForm = widgetContainer.find( '> .widget-inside > .form, > .widget-inside > form' ); // Note: '.form' appears in the customizer, whereas 'form' on the widgets admin screen.
+		widgetContent = widgetForm.find( '> .widget-content' );
+
+		idBase = widgetForm.find( '> .id_base' ).val();
+		if ( 'text' !== idBase ) {
+			return;
+		}
+
+		widgetId = widgetForm.find( '> .widget-id' ).val();
+
+		// @todo Obtain the Save/Apply button and detatch the TinyMCE container element upon clicking, to then restore upon widgetUpdated.
+
+		// Prevent initializing already-added widgets.
+		if ( component.widgetControls[ widgetId ] ) {
+			return;
+		}
+
+		widgetControl = new component.TextWidgetControl({
+			el: widgetContent
+		});
+
+		component.widgetControls[ widgetId ] = widgetControl;
+
+		/*
+		 * Render the widget once the widget parent's container finishes animating,
+		 * as the widget-added event fires with a slideDown of the container.
+		 * This ensures that the textarea is visible and an iframe can be embedded
+		 * with TinyMCE being able to set contenteditable on it.
+		 */
+		widgetInside = widgetContainer.parent();
+		renderWhenAnimationDone = function() {
+			if ( widgetInside.is( ':animated' ) ) {
+				setTimeout( renderWhenAnimationDone, animatedCheckDelay );
+			} else {
+				widgetControl.render();
+			}
+		};
+		renderWhenAnimationDone();
+	};
+
+	/**
+	 * Sync widget instance data sanitized from server back onto widget model.
+	 *
+	 * This gets called via the 'widget-updated' event when saving a widget from
+	 * the widgets admin screen and also via the 'widget-synced' event when making
+	 * a change to a widget in the customizer.
+	 *
+	 * @param {jQuery.Event} event - Event.
+	 * @param {jQuery}       widgetContainer - Widget container element.
+	 * @returns {void}
+	 */
+	component.handleWidgetUpdated = function handleWidgetUpdated( event, widgetContainer ) {
```

Comment:
So I was trying to figure out the flow of things here a bit and added some logging statements to this method - and I was never able to trigger this logic.  In an older version of this branch, it seemed this logic was called on any `change` event in MCE but for now I can't seem to make it happen.